vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/gemini/post_account/post_account_gemini_unittest.cc

```diff
@@ -0,0 +1,133 @@
+  std::unique_ptr<PostAccount> post_account_;
+
+  GeminiPostAccountTest() {
+    mock_ledger_client_ = std::make_unique<ledger::MockLedgerClient>();
+    mock_ledger_impl_ =
+        std::make_unique<ledger::MockLedgerImpl>(mock_ledger_client_.get());
+    post_account_ = std::make_unique<PostAccount>(mock_ledger_impl_.get());
+  }
+};
+
+TEST_F(GeminiPostAccountTest, ServerOK) {
+  ON_CALL(*mock_ledger_client_, LoadURL(_, _))
+      .WillByDefault(Invoke(
+          [](type::UrlRequestPtr request, client::LoadURLCallback callback) {
+            type::UrlResponse response;
+            response.status_code = 200;
+            response.url = request->url;
+            response.body = R"({
+              "account": {
+                "accountName": "Primary",
+                "shortName": "primary",
+                "type": "exchange",
+                "created": "1619040615242",
+                "verificationToken": "mocktoken"
+              },
+              "users": [{
+                "name": "Test",
+                "lastSignIn": "2021-04-30T18:46:03.017Z",
+                "status": "Active",
+                "countryCode": "US",
+                "isVerified": true
+              }],
+              "memo_reference_code": "GEMAPLLV"
+            })";
+            callback(response);
+          }));
+
+  post_account_->Request(
+      "4c2b665ca060d912fec5c735c734859a06118cc8",
+      [](const type::Result result, const std::string address,
+         const std::string linking_info, const std::string user_name,
+         const bool verified) {
+        EXPECT_EQ(result, type::Result::LEDGER_OK);
+        EXPECT_EQ(address, "Primary");
+        EXPECT_EQ(linking_info, "mocktoken");
+        EXPECT_EQ(user_name, "Test");
+        EXPECT_TRUE(verified);
+      });
+}
+
+TEST_F(GeminiPostAccountTest, ServerError401) {
+  ON_CALL(*mock_ledger_client_, LoadURL(_, _))
+      .WillByDefault(Invoke(
+          [](type::UrlRequestPtr request, client::LoadURLCallback callback) {
+            type::UrlResponse response;
+            response.status_code = 401;
+            response.url = request->url;
+            response.body = "";
+            callback(response);
+          }));
+
+  post_account_->Request(
+      "4c2b665ca060d912fec5c735c734859a06118cc8",
+      [](const type::Result result, const std::string address,
+         const std::string linking_info, const std::string user_name,
+         const bool verified) {
+        EXPECT_EQ(result, type::Result::EXPIRED_TOKEN);
+        EXPECT_EQ(address, "");
+        EXPECT_EQ(linking_info, "");
+        EXPECT_EQ(user_name, "");
+        EXPECT_FALSE(verified);
+      });
+}
+
+TEST_F(GeminiPostAccountTest, ServerErrorRandom) {
+  ON_CALL(*mock_ledger_client_, LoadURL(_, _))
+      .WillByDefault(Invoke(
+          [](type::UrlRequestPtr request, client::LoadURLCallback callback) {
+            type::UrlResponse response;
+            response.status_code = 453;
```

Comment:
I doubt 453 will be used by GeminiPostAccount but `418 I'm A Teapot` could be safer for a random response status code

Reply:
This error code was an April Fools joke, so a good candidate

Reply:
There should also be tests to cover all status codes in `PostAccount::CheckStatusCode`

Reply:
`PostAccount::ParseBody` should ideally be added to cover the optional and required nodes. Preferably by decoupling parsing from post account.

Reply:
Above points would apply to all unit tests for this PR

Reply:
resolved